reactor-core/src/main/java/reactor/core/publisher/Sinks.java

```diff
@@ -203,64 +159,43 @@ public void error(Throwable e) {
 		}
 
 		@Override
-		public StandaloneFluxSink<T> next(T t) {
+		public Sink<T> next(T t) {
 			delegateSink.next(t);
 			return this;
 		}
 	}
 
-	//TODO improve synchronization, prefer CAS ?
-	static final class MonoProcessorSink<T> implements StandaloneMonoSink<T> {
+	static final class MonoProcessorSink<T> implements Sink<T> {
```

Comment:
if sinks are advertised as thread safe objects, doesn't this require external synchronization of the processor's calls ? especially if we later introduce several flavors of `MonoProcessor` ?

Reply:
MonoProcessor are safe, all the onXxx methods are atomic - I'll actually have an extra commit that moves further and remove the need for MonoProcessorSink by making MonoProcessor implementing it directly.

Reply:
ok. the later might be a better solution actually, since then there is no risk of adding a (non-serialized) flavor of `MonoProcessor` and mistakenly wrapping it with a (non-serialized) sink decorator